pkg/devcontainer/feature/features.go

```diff
@@ -241,11 +242,22 @@ func downloadFeatureFromURL(url string, destFile string, log log.Logger) error {
 
 	// initiate download
 	log.Debugf("Download feature from %s", url)
-	resp, err := devpodhttp.GetHTTPClient().Get(url)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return errors.Wrap(err, "make request")
+	}
+	for key, value := range httpHeaders {
+		req.Header.Set(key, value)
```

Comment:
Could you ensure we're not accidentally overwriting existing headers here please?

Reply:
Do you mean I shall check for duplicates in the json file?
Because the req is created in the method, so it can't have any headers by default.

For the duplicate keys in the json it is going to be complicated. According to this https://github.com/golang/go/issues/48298, JSON unmarshaller is not able to deal with map containing duplicated keys.

However, the author states that this could be solved in json v2 package, so the solution would be to update the dependency in the devpod. If we still want to do this in devpod it means to manually parse the bytes in a specific unmarshall function for the features to detect duplicates just to display a warning. Not sure that it is worth the cost.

What do you think?

Reply:
> Because the req is created in the method, so it can't have any headers by default.

Ah yeah right, dismiss what I said earlier.
In the case of duplicate keys in `devcontainer.json` it's a configuration error and doesn't need to be handled by DevPod imo